kf_lib_data_ingest/etl/extract/extract.py

```diff
@@ -185,15 +189,31 @@ def _run(self):
                                             ).set_index('index')
             del df_out.index.name
 
-            # standardize on string values
-            df_out = df_out.astype('object')
-            df_out[df_out.notnull()] = df_out[df_out.notnull()].astype('str')
+            # standardize values again after operations
+            df_out = self._clean_up_df(df_out)
 
             output[extract_config.config_filepath] = (data_path, df_out)
 
         # return dictionary of all dataframes keyed by extract config paths
         return output
 
+    def _clean_up_df(self, df):
+        # We can't universally control which null type will get used by a data
+        # file loader, and it might also change, so let's always push them all
+        # to empty strings because nulls are not our friends. It's easier for a
+        # configurator to equate empty spreadsheet cells with empty strings.
+
+        # Typed loaders like pandas.read_json force us into storing numerically
+        # typed values, and then nulls, which read_json does not let you handle
+        # inline, cause pandas to convert perfectly good ints into ugly floats.
+        # So here we get any untidy values back to nice and tidy strings.
+
+        return df.applymap(
+            lambda x: to_str_with_floats_downcast_to_ints_first(
+                x, replace_na=True, na=''
```

Comment:
Why use empty strings here instead of None? If you leave the empty strings then transform will have to convert all empty strings to None later on anyway when its doing null processing.  Might as well do it here since, the output of _clean_up_df basically goes into transform

Reply:
We just have to decide whether someone writing an extract configuration will look for "" or for None. Either way is ok.

Reply:
Yea None is better